dev/com.ibm.ws.security.oidc.social_fat.backchannelLogout/fat/src/com/ibm/ws/security/oidc_social/backchannelLogout/fat/LogoutTokenCreationTests.java

```diff
@@ -0,0 +1,547 @@
+                    boolean found = false;
+                    for (String logoutTokenValue : logoutTokenList) {
+                        if (logoutTokenValue.equals(idTokenValue)) {
+                            found = true;
+                        }
+                    }
+                    if (!found) {
+                        fail("Member: " + idTokenValue + " was not found in the logout_token - it is not valid.");
+                    }
+                }
+
+            }
+        }
+
+    }
+
+    protected void mustHaveOneOrTheOtherString(String oneKey, String otherKey, Map<String, Object> idTokenDataClaims, Map<String, Object> logoutClaims) throws Exception {
+
+        Object oneLogoutTokenValue = logoutClaims.get(oneKey);
+        Object otherLogoutTokenValue = logoutClaims.get(otherKey);
+
+        if (oneLogoutTokenValue == null) {
+            if (otherLogoutTokenValue == null) {
+                // have to have one or the other or both - having neither is NOT allowed
+                fail("Both key: " + oneKey + " and key: " + otherKey + " were missing in the logout_token");
+            }
+        } else {
+            // validate against what's in the id_token
+            mustHaveString(oneKey, idTokenDataClaims, logoutClaims, true);
+        }
+
+        // don't have to check for null - if it's null, we don't need to check it and we've already checked for both claims being null
+        if (otherLogoutTokenValue != null) {
+            // validate against what's in the id_token
+            mustHaveString(otherKey, idTokenDataClaims, logoutClaims, true);
+        }
+
+    }
+
+    protected void mustNotHave(String key, Map<String, Object> logoutClaims) throws Exception {
+
+        Object logoutTokenValue = logoutClaims.get(key);
+        if (logoutTokenValue != null) {
+            fail("Key: " + key + " was included in the logout_token and should NOT have been");
+        }
+    }
+
+    protected void mustHaveEvents(Map<String, Object> logoutClaims) throws Exception {
+
+        Object rawLogoutTokenValue = logoutClaims.get(Constants.PAYLOAD_EVENTS);
+        if (rawLogoutTokenValue == null) {
+            fail("Key: events is missing in the logout_token");
+        }
+
+        if (rawLogoutTokenValue instanceof JsonObject) {
+
+            Log.info(thisClass, "mustHaveEvents", "JsonObject: " + rawLogoutTokenValue);
+            JSONObject event = (JSONObject) rawLogoutTokenValue;
+            if (!event.containsKey(logoutEventKey)) {
+                fail("Key: events in the logout_token does not contain a Json object containing the key: " + logoutEventKey);
+            }
+            Object value = event.get(logoutEventKey);
+            if (value == null) {
+                // should really be able to have a null content, but, just in case
+                fail(logoutEventKey + " value is null and shouldcontain an empty Json object");
+            } else {
+                if (!value.equals(new JSONObject())) {
```

Comment:
```suggestion
            } else {
                // There needs to be validation added here to make sure `value` is JSON. The ensuing if statement is fine as-is since it'll log the value if it's not the empty JSON object like we talked about. But with the code as it is here, if `value` is `"some string"` or `123`, all this will do is log that `value` wasn't equal to an empty JSON object when it should fail because `value` isn't JSON in the first place.
                if (!value.equals(new JSONObject())) {
```